os/hal/ports/STM32/STM32F4xx/hal_lld.c

```diff
@@ -64,15 +64,17 @@ static void hal_lld_backup_domain_init(void) {
   }
 
 #if STM32_LSE_ENABLED
+  int rusefiLseCounter = 0;
 #if defined(STM32_LSE_BYPASS)
   /* LSE Bypass.*/
   RCC->BDCR |= RCC_BDCR_LSEON | RCC_BDCR_LSEBYP;
 #else
   /* No LSE Bypass.*/
   RCC->BDCR |= RCC_BDCR_LSEON;
 #endif
-  while ((RCC->BDCR & RCC_BDCR_LSERDY) == 0)
-    ;                                       /* Waits until LSE is stable.   */
+  while (rusefiLseCounter++ < RUSEFI_STM32_LSE_WAIT_MAX
```

Comment:
do we want to change the value of RTCSEL based on whether or not HSE works?

Reply:
I assume you mean LSE (vs HSE)?

Reviewing the code, it seems like that's important, however the logic in ChibiOS 16 patch doesn't do this: I guess rusEFI is changing the RTC source already, or this oversight is part of the/a bug that's existed then.

Reply:
sorry I mean LSE - right now it'll use the value of RTCSEL blindly even if LSE doesn't work, when it should instead use LSI

Reply:
I implemented a fallback as it seems necessary.  I believe without it the CPU resets, but unsure (all LEDs other than Power dimly lit).

Reply:
It doesn't reset, it just hangs during init of some step that relies on the oscillator being there, which it isn't.

Reply:
Ah okay.  Seems the mode is the same as without this patch (but STM32_LSE_ENABLED; waiting in the busy-loop).  I wonder what causes the LEDs to be dim, but no matter.

Reply:
with a debugger you can just break in and see where it's hung

Reply:
Debugger comes next week :+1:.